include/vpnes/core/device.hpp

```diff
@@ -520,7 +520,7 @@ class CEventDevice : public CClockedDevice {
 	 */
 	void registerDeviceEvent(CEvent *event) {
 		assert(m_EventData.find(event) == m_EventData.end());
-		auto newData = m_EventData.emplace(event, event);
+		auto newData = m_EventData.emplace(event, SEventData(event));
```

Comment:
`std::is_constructable<SEventData, CEvent *>::value` should definitely be `true`.